corehq/apps/user_importer/importer.py

```diff
@@ -659,21 +660,33 @@ def create_or_update_web_users(upload_domain, user_specs, upload_user, update_pr
                 else:
                     membership = user.get_domain_membership(domain)
                     if membership:
-                        if domain_info.can_assign_locations and location_codes is not None:
-                            location_ids = find_location_id(location_codes, domain_info.location_cache)
-                            locations_updated, primary_loc_removed = check_modified_user_loc(location_ids,
-                                                                                             membership.location_id,
-                                                                                             membership.assigned_location_ids)
-                            if primary_loc_removed:
-                                user.unset_location(domain, commit=False)
-                            if locations_updated:
-                                user.reset_locations(domain, location_ids, commit=False)
-                        user_current_role = user.get_role(domain=domain)
-                        role_updated = not (user_current_role
-                                            and user_current_role.get_qualified_id() == role_qualified_id)
-                        if role_updated:
-                            user.set_role(domain, role_qualified_id)
-                        user.save()
+                        def _modify_existing_user_in_domain(current_user, max_tries=3):
```

Comment:
can this be converted to a top level function?

Reply:
[61cbdce](https://github.com/dimagi/commcare-hq/pull/28832/commits/61cbdce72fecac75ecabccd8fbefe8df8c669ce4)